spec/core/ics-004-channel-and-packet-semantics/README.md

```diff
@@ -823,37 +869,64 @@ function timeoutPacket(
     abortTransactionUnless(provableStore.get(packetCommitmentPath(packet.sourcePort, packet.sourceChannel, packet.sequence))
            === hash(packet.data, packet.timeoutHeight, packet.timeoutTimestamp))
 
-    if channel.order === ORDERED {
-      // ordered channel: check that packet has not been received
-      abortTransactionUnless(nextSequenceRecv <= packet.sequence)
-      // ordered channel: check that the recv sequence is as claimed
-      abortTransactionUnless(connection.verifyNextSequenceRecv(
-        proofHeight,
-        proof,
-        packet.destPort,
-        packet.destChannel,
-        nextSequenceRecv
-      ))
-    } else
-      // unordered channel: verify absence of receipt at packet index
-      abortTransactionUnless(connection.verifyPacketReceiptAbsence(
-        proofHeight,
-        proof,
-        packet.destPort,
-        packet.destChannel,
-        packet.sequence
-      ))
+    switch channel.order {
+      case ORDERED:
+        // ordered channel: check that packet has not been received
+        abortTransactionUnless(nextSequenceRecv <= packet.sequence)
```

Comment:
Is not related to the `ORDERED_ALLOW_TIMEOUT` channels, but shouldn't this condition be `abortTransactionUnless(nextSequenceRecv == packet.sequence)`? Why do we allow timeout messages regarding future packets. Maybe the packet with  `nextSequenceRecv` will still be received on time, even if another packet with e.g.,  `nextSequenceRecv+1` has timed out.

Reply:
Hmm yea that makes sense to me.